contracts/identity/Identity.sol

```diff
@@ -294,3 +294,374 @@ contract Identity is IdentityAdminRole, SchemeGuard, Pausable {
+     */
+    function _addWhitelistedWithDID(address account, string memory did) internal {
+        bytes32 pHash = keccak256(bytes(did));
+        require(didHashToAddress[pHash] == address(0), "DID already registered");
+
+        addrToDID[account] = did;
+        didHashToAddress[pHash] = account;
+
+        _addWhitelisted(account);
+    }
+
+
+    /* @dev Internal function to remove from whitelisted
+     * @param account the address to add
+     */
+    function _removeWhitelisted(address account) internal {
+        whitelist.remove(account);
+
+        whitelistedCount -= 1;
+        delete dateAuthenticated[account];
+        whitelistDateUpdated[account] = now;
+
+        if (isContract(account)) {
+            whitelistedContracts -= 1;
+        }
+
+        string memory did = addrToDID[account];
+        bytes32 pHash = keccak256(bytes(did));
+
+        delete dateAuthenticated[account];
+        delete addrToDID[account];
+        delete didHashToAddress[pHash];
+
+        emit UpdatedLastAuthenticated(account);
+        emit WhitelistedRemoved(account);
+    }
+    /* @dev Returns true if given address has been added to the blacklist
+     * @param account the address to check
+     * @return a bool indicating weather the address is present in the blacklist
+     */
+      function _isBlacklisted2(address account) internal view returns (bool) {
+        return blacklist.has(account);
+    }
+
+    /* @dev Returns true if given address has been added to the blacklist
+     *  First checks if it was updated in the current contract and if not,
+     *  returns the result of old contract
+     * @param account the address to check
+     * @return a bool indicating weather the address is present in the blacklist
+     */
+    function isBlacklisted(address account) public view returns (bool) {
+        if (blacklistDateUpdated[account] > 0){
+            return ;}
+        return identity.isBlacklisted(account);
+    }
+
+    /* @dev Internal function to see if given address is a contract
+     * @return true if address is a contract
+     */
+    function _isContract2(address _addr) internal view returns (bool) {
+        uint256 length;
+        assembly {
+            length := extcodesize(_addr)
+        }
+        return length > 0;
+    }
+    /* @dev Function to see if given address is a contract.
+     *  First checks if it was updated in the current contract and if not,
+     *  returns the result of old contract
+     * @return true if address is a contract
+     */
+    function isContract(address _addr) internal view returns (bool) {
+        return (identity.isContract(_addr) || _isContract2(_addr));
+    }
+
+    /* @dev Function to update DID, only by the account owner
+     * @param account the address of the account owner
+     * @param did the new DID to update
+     */
+    function updateDID(address account, string memory did)
+      public
+      onlyAccountOwner
```

Comment:
this is missing something

Reply:
I'm missing the parameter for the modifier, it should be:

function updateDID(address account, string memory did)
      public
      onlyAccountOwner(account)